osu.Game.Rulesets.Taiko/Difficulty/Skills/Rhythm.cs

```diff
@@ -13,7 +13,7 @@ namespace osu.Game.Rulesets.Taiko.Difficulty.Skills
     /// <summary>
     /// Calculates the rhythm coefficient of taiko difficulty.
     /// </summary>
-    public class Rhythm : Skill
+    public class Rhythm : StrainSkill
```

Comment:
This skill is special (it specifies `StrainDecayBase => 0`, as the goal is to have this one be based on object index, not time between objects). I'd ideally want that zero specification gone from this to avoid having to explain it, so in that regard I don't think this change goes far enough.

Reply:
I agree, we should ideally not have that `StrainDecayBase => 0` in the skill (and in fact the whole skill is pretty hacky to just fit into the old `Skill` requirements), however I don't think this is the place to make that change.
I think the most elegant solution would be to split `StrainSkill` again perhaps into `StrainSkill` and `TimeStrainSkill` or something since the current strain implementation is centered around consistent time-based decay, where as the taiko `Rhythm` skill is fundamentally different in that it has different decay requirements and forcefully resets the strain to 0 in places.

Reply:
If the strain skill split was its own separate pull then it would be the best place to make this change, but clumped up like this it's definitely not a good idea to pile more on.

Reply:
Yea I agree. I'm happy to start work on that refactor as a new PR following this one.